library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -206,8 +206,8 @@ DownloadFileId id() {
         return downloadFileStatus.downloadFileId();
     }
 
-    boolean matches(String networkUrl) {
-        return networkUrl.equals(url);
+    boolean matches(DownloadFileId downloadFileId) {
+        return downloadFileStatus.downloadFileId().equals(downloadFileId);
```

Comment:
this class could hold a `downloadFileId`

Reply:
https://github.com/novoda/download-manager/pull/292/commits/4e16f8509a1fed9063ec48db77b76c246e9cf71b